ckanext/canada/tests/factories.py

```diff
@@ -1,12 +1,76 @@
-from ckan.tests.factories import Organization
+# -*- coding: UTF-8 -*-
+from ckan.tests.factories import User, Organization, Dataset, Resource, ResourceView
+from factory import LazyAttribute
+import ckan.tests.helpers as helpers
+from ckan.model import Session
+
+
+class CanadaUser(User):
+    @classmethod
+    def _create(self, target_class, *args, **kwargs):
+        if args:
+            assert False, "Positional args aren't supported, use keyword args."
+
+        # The ckan.tests.factories.User does not add the user to a session
+        # so we need to override the _create method to add the user to the 
+        # session to prevent DetachedInstanceError execptions from being raised
+        user = target_class(**dict(kwargs, sysadmin=False))
+        Session.add(user)
```

Comment:
This is really odd, but we can revisit it in the 2.9 branch after switching to pytest

Reply:
Yes it is. Looks like the factories are all different in 2.9+ so hopefully it will just work normally in there.